extensions/panache/mongodb-panache/deployment/src/main/java/io/quarkus/mongodb/panache/deployment/PanacheResourceProcessor.java

```diff
@@ -77,6 +81,25 @@ ReflectiveHierarchyBuildItem registerForReflection(CombinedIndexBuildItem index)
         return new ReflectiveHierarchyBuildItem(type, compositeIndex);
     }
 
+    public static final DotName MONGOCLIENT_INTERFACE = DotName
+            .createSimple(com.mongodb.client.MongoClient.class.getName());
+
+    @BuildStep
+    void unremovableProducers(BuildProducer<UnremovableBeanBuildItem> unremovable,
+            BeanArchiveIndexBuildItem beanArchiveIndex) {
+        Type type = Type.create(MONGOCLIENT_INTERFACE, Type.Kind.CLASS);
+        unremovable.produce(
+                new UnremovableBeanBuildItem(new Predicate<BeanInfo>() {
+                    @Override
+                    public boolean test(io.quarkus.arc.processor.BeanInfo beanInfo) {
+                        io.quarkus.arc.processor.BeanInfo declaringBean = beanInfo.getDeclaringBean();
+                        return (beanInfo.isProducerMethod() || beanInfo.isProducerField())
```

Comment:
Is this complex dance actually needed? Won't it suffice to have the actual Mongo class be an unremovable bean?

Reply:
I have simplified by removing field part and use equal on mongoclient producer to be sure that we handle only the class we need. currently it mark all MongoClient producer (not only default, named one too). I think we have to keep it because, we can add in future a system to handle multi db on panache mongo